### DISCUSSION TITLE
Close a requested item

### DISCUSSION THREAD

**Discussion by fbcarvalho** [Snow-Shell/servicenow-powershell#144] [Q&A] [answered]
How do I close a requested item using the powershell?

**Comment by gdbarron** [comment, marked as answer]
Hi @fbcarvalho.  Closing a requested item typically involves closing all the attached catalog items and then the requested item automatically closes.  This is going to get a lot easier in the next release I'm currently working on, but for now with 2.x you can perform the following:

``` powershell
# get the requested item sys_id
$sysId = Get-ServiceNowRecord -Table 'Requested Item' -Filter @('number','-eq','ritmxxxxxx') -Property sys_id | Select -ExpandProperty sys_id # replace ritmxxxxxx with your actual number

# set catalog tasks to closed complete
Get-ServiceNowRecord -Table sc_task -Filter @('request_item', '-eq', $sysId) | Update-ServiceNowRecord -Table sc_task -Values @{'state'='Closed Complete'}
```

You may or may not have additional fields that must be populated when closing a task.  If so, you'll need to add those values to `-Values` as well.  You can determine which are required by closing a task via the GUI and seeing if it prompts you.

**Reply by fbcarvalho** [reply]
Thank you so much.
I'm working to automate some resquested items using System Center Orchestrator and powershell scripts. So firts I need to close all tasks and consequently the RITM will be closed automatically.
